### PR TITLE
jsonrpc security, cors headers, fixed #359

### LINKED ISSUE #359: JSONRPC security
Allow management of which hosts are allowed to connect to the JSONRPC interface.

### PR DESCRIPTION
**Pull request by debris**

### CODE REVIEW

**Comment by gavofyork** [comment]
is this just a bit of http-level security or does it actually prevent connections from particular domains? (typical example is that you only want to allow localhost connections)

File: parity/main.rs [hunk 1]
```diff
@@ -81,11 +81,12 @@ Options:
 
   -j --jsonrpc             Enable the JSON-RPC API sever.
   --jsonrpc-url URL        Specify URL for JSON-RPC API server [default: 127.0.0.1:8545].
+  --jsonrpc-cors URL       Specify CORS header for JSON-RPC API responses [default: null].
```
Comment (arkpar) [inline parity/main.rs]: It is not clear what does default "null" mean?  No header?

Comment (debris) [inline parity/main.rs]: "null" means that `Access-Control-Allow-Origin` value will be `null`
https://www.w3.org/TR/cors/#access-control-allow-origin-response-header

**Comment by debris** [comment]
> is this just a bit of http-level security or does it actually prevent connections from particular domains? (typical example is that you only want to allow localhost connections)

It's just http-level security, the same that is in cpp/go clients.

**Comment by gavofyork** [comment]
is it all that go clients provide? i seem to remember it had two separate "json rpc" security settings.

**Comment by debris** [comment]
that's all.

full list of geth api options:

```
API AND CONSOLE OPTIONS:
  --rpc									Enable the HTTP-RPC server
  --rpcaddr "127.0.0.1"							HTTP-RPC server listening interface
  --rpcport "8545"							HTTP-RPC server listening port
  --rpcapi "db,eth,net,web3"						API's offered over the HTTP-RPC interface
  --ipcdisable								Disable the IPC-RPC server
  --ipcapi "admin,db,eth,debug,miner,net,shh,txpool,personal,web3"	API's offered over the IPC-RPC interface
  --ipcpath "/Users/marek/Library/Ethereum/geth.ipc"			Filename for IPC socket/pipe
  --rpccorsdomain 							Domains from which to accept cross origin requests (browser enforced)
  --jspath "."								JavaSript root path for `loadScript` and document root for `admin.httpGet`
  --exec 								Execute JavaScript statement (only in combination with console/attach)
```

**Pull request merged**